sourcecode/report_generator.py

```diff
@@ -0,0 +1,167 @@
+import calendar
+
+
+class ReportGenerator():
+    def generate_report_year(report: dict) -> None:
+        """
+        Used for priniting report
+        
+        @params
+        :report dict: contains values to be printed
+        """
+        highest_temp = report['highest_temp']
+        highest_temp_date = report['highest_temp_date']
+        lowest_temp = report['lowest_temp']
+        lowest_temp_date = report['lowest_temp_date']
+        humidity = report['humidity']
+        humidity_date = report['humidity_date']
+
+        print (f'Highest: {highest_temp}C on {highest_temp_date}')
```

Comment:
Can't we directly use `report` dict in fstring instead of declaring variables and compromising space?

Reply:
Same thing in other functions as well.